compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java

```diff
@@ -946,6 +946,17 @@ private void createLocalVariableTable(BIRFunction func, BIRVarToJVMIndexMap inde
                 if (localVar.endBB != null) {
                     endLabel = labelGen.getLabel(funcName + endBB.id.value + "beforeTerm");
                 }
+            } else if (localVar.kind == VarKind.SYNTHETIC && isCompilerAddedVars(localVar.metaVarName)) {
+                if (localVar.startBB != null) {
+                    startLabel = labelGen.getLabel(funcName + SCOPE_PREFIX + localVar.insScope.id);
+                }
+                for (int j = i + 1; j < func.localVars.size(); j++) {
```

Comment:
does this var not have a `endBB`? If so, isn't that an issue.

Reply:
We need to end the visibility of inner variables once we came out the expression. `Let` expression does not have any terminator instructions and so we can't keep the end label based on the `endBB`. 
eg: For the following ballerina code, 
```
public function main() {

    int a = 1;

    string greeting = let string hello = "Hello ", string ballerina = "Ballerina!" 
                        in hello + ballerina;

    int b = 5;
}
```
BBs are generated like this,
```
bb0 {
        %1 = ConstLoad 1;
        %4 = ConstLoad Hello ;
        %6 = ConstLoad Ballerina!;
        %8 = %4 + %6;
        %3 = %8;
        %13 = ConstLoad 5;
        %0 = ConstLoad 0;
        GOTO bb2;
    }
    bb1 {
        GOTO bb2;
    }
    bb2 {
        return;
    }
```

Reply:
I believe the correct thing to do is to spit `bb0` to two bb's in the frontend (we also have to make sure our optimizer doesn't merge them)
eg:
```
    bb0 {
        %1 = ConstLoad 1;
        %4 = ConstLoad Hello ;
        %6 = ConstLoad Ballerina!;
        %8 = %4 + %6;
        %3 = %8;
    }
    bb1 {
        %13 = ConstLoad 5;
        %0 = ConstLoad 0;
        GOTO bb3;
    }
    bb2 {
        GOTO bb3;
    }
    bb3 {
        return;
    }
```

then we can use above `bb1` as the end bb.